examples/src/Examples/UseRef.purs

```diff
@@ -0,0 +1,34 @@
+module Examples.UseRef
+  ( view
+  ) where
+
+import Prelude
+
+import Data.Foldable (traverse_)
+import Data.Maybe (Maybe(..))
+import Elmish (ReactElement)
+import Elmish.HTML.Styled as H
+import Elmish.Hooks (withHooks)
+import Elmish.Hooks.UseRef (readRef, useRef)
+import Unsafe.Coerce (unsafeCoerce)
+import Web.HTML.HTMLElement (focus)
+
+view :: ReactElement
+view = withHooks do
+  inputEl <- useRef Nothing
+  let onButtonClick = traverse_ focus =<< readRef inputEl
+  pure $
+    H.div "row mt-3"
+    [ H.div "col-12 col-md-6 col-lg-4"
+      [ H.h2 ""
+        [ H.code "" "useRef"
+        , H.text " hook"
+        ]
+      , H.div "row"
+        [ H.div "col" $
+            H.input_ "form-control" { ref: unsafeCoerce inputEl, defaultValue: "" }
```

Comment:
I don't really understand how you got this to compile. The `ref` prop is explicitly banned at the source: https://github.com/collegevine/purescript-elmish/blob/66d6f91c037eb55b0c5011f709a74335ec7936c1/src/Elmish/React.purs#L109

We really need to remove that, it doesn't serve anything.

But as long as you did manage to get it to compile, did you know that `ref` doesn't have to be a record `{ current: foo }`, but can also be a function? And then this function just gets called on every render with the element as parameter. So with some careful deduping, this could be achieved just with `useState`:

```
withHooks do
  el /\ setEl <- useState Nothing
  ...
  H.input_ "" 
    { ref: setEl <?| \r -> if eqByReference r <$> el == Just true then Nothing else Just (Just el) 
    }
```

Where `eqByReference` is a function that just wraps JS `===` comparison operator. It's necessary to check and avoid calling `setEl` for the same element, because it happens on every render, so might create an infinite loop. 

Although it might actually work with `useState` specifically due to how it stores its state. But better safe than sorry.

Reply:
> The ref prop is explicitly banned at the source

That's only for imported react components. I don't know why it's disallowed, maybe because you have to pass the ref to an actual DOM node? But that's what I'm doing: passing it to an Elmish HTML component (which allows a `ref :: String` prop) and coercing to a `String`.

I did forget you can pass a function, but just copied the way the hook works in React. It's documented as a general purpose mutable reference, but I wonder if it's really useful for more than DOM elements.

What do you mean by "better safe than sorry?"

Reply:
Elmish HTML combinators do use the same importing mechanism, so the `ref` prop should be banned for them too. I just tried it out myself, and I do indeed get the error message. I don't understand how it works for you.

By "better safe than sorry" I mean: when using `useState`, the infinite loop may not happen (not sure, need to confirm) due to how `wrapWithLocalState` stores state, and therefore `eqByReference` guard may be unnecessary. But even if that turns out to be the case, I still think that the guard should be used.